lib/service/service.go

```diff
@@ -3259,6 +3268,48 @@ func (process *TeleportProcess) initDiagnosticService() error {
 	return nil
 }
 
+// initDebugService starts debug service serving endpoints used for
+// troubleshooting the instance.
+func (process *TeleportProcess) initDebugService() error {
+	logger := process.logger.With(teleport.ComponentKey, teleport.Component(teleport.ComponentDebug, process.id))
+
+	listener, err := process.importOrCreateListener(ListenerDebug, filepath.Join(process.Config.DataDir, debug.ServiceSocketName))
+	if err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
I retested this error + process forking. I've made some changes to the `createListener` function because for unix sockets, we need to:
- Do not remove the file descriptor when the listener is closed. Otherwise, the child process would be handed a listener that doesn't have a socket file.
- Ensure the file is not present before starting it (due to it not being deleted).

I've added comments with the motivations, and there is also relevant discussion that mentions a similar mechanism we have here, and their considerations: https://groups.google.com/g/Golang-nuts/c/UtBR4IfgaEw